processor/k8sprocessor/kube/client.go

```diff
@@ -93,7 +99,12 @@ func New(logger *zap.Logger, apiCfg k8sconfig.APIConfig, rules ExtractionRules,
 		newInformer = newSharedInformer
 	}
 
+	if newNamespaceInformer == nil {
+		newNamespaceInformer = newNamespaceSharedInformer
+	}
+
 	c.informer = newInformer(c.kc, c.Filters.Namespace, labelSelector, fieldSelector)
+	c.namespaceInformer = newNamespaceInformer(c.kc)
```

Comment:
Looks like we will instantiate a namespace informer irrespective of whether a user configured the processor to extract any labels/annotations from namespaces or not. I think we should do it only if the processor actually need to use it.